<launch>
  <!-- SE2 Controller for X-ray Source -->
  
  <!-- Load the SE2 controller nodelet -->
  <node pkg="nodelet" type="nodelet" name="se2_control_nodelet"
        args="load se2_control/SE2ControlNodelet /xray_source"
        output="screen">
    
    <!-- Vehicle parameters -->
    <param name="vehicle_name" value="xray_source" />
    <param name="mass" value="15.0" />
    <param name="wheel_base" value="0.46" />
    <param name="track" value="0.412" />
    <param name="wheel_radius" value="0.08" />
    
    <!-- Control gains -->
    <param name="kx_x" value="1.0" />
    <param name="kx_y" value="1.0" />
    <param name="kyaw" value="1.0" />
    
    <!-- Initial position (optional) -->
    <param name="init_x" value="0.0" />
    <param name="init_y" value="0.0" />
    <param name="init_z" value="-9999.0" />
    
    <!-- Use external yaw -->
    <param name="use_external_yaw" value="true" />
  </node>

  <!-- Optional: Remap topics for compatibility -->
  <node pkg="topic_tools" type="relay" name="cmd_vel_relay"
        args="/xray_source/cmd_vel /xray_source/four_wheel_steering_command" />
        
</launch> 